newrelic/resource_newrelic_cloud_gcp_link_account.go

```diff
@@ -0,0 +1,148 @@
+package newrelic
+
+import (
+	"context"
+	"strconv"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/newrelic/newrelic-client-go/pkg/cloud"
+)
+
+func resourceNewRelicCloudGcpLinkAccount() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceNewRelicCloudGcpLinkAccountCreate,
+		ReadContext:   resourceNewRelicCloudGcpLinkAccountRead,
+		UpdateContext: resourceNewRelicCloudGcpLinkAccountUpdate,
+		DeleteContext: resourceNewRelicCloudGcpLinkAccountDelete,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "name of the linked account",
+				Required:    true,
+			},
+			"project_id": {
+				Type:        schema.TypeString,
+				Description: "project id of the Gcp account",
+				Required:    true,
+			},
+		},
+	}
+}
+
+func resourceNewRelicCloudGcpLinkAccountCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	providerConfig := meta.(*ProviderConfig)
+	client := providerConfig.NewClient
+	accountID := selectAccountID(providerConfig, d)
+	linkAccountInput := expandGcpCloudLinkAccountInput(d)
+	cloudLinkAccountPayload, err := client.Cloud.CloudLinkAccountWithContext(ctx, accountID, linkAccountInput)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+	var diags diag.Diagnostics
+	if len(cloudLinkAccountPayload.Errors) > 0 {
```

Comment:
I wonder if we'll have the same problem as the AWS provider. There the initial connection failed because the resources weren't created quickly enough in AWS, but it did succeed on next attempt. Maria added some retry code to AWS to handle that case: https://github.com/newrelic/terraform-provider-newrelic/pull/1608/files#diff-e7f2180b00f9dc0400207dc507066a63d1b515cee7da2daf4acabd59f12d2ca5R63 

If you haven't already I would create a full end to end example (can be a new story) which the customer could use to fully automatically set up the integration. If this is a problem you will notice if you run that.